contracts/src/lido/LidoGatewayManager.sol

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.16;
+
+import {ScrollGatewayBase} from "../libraries/gateway/ScrollGatewayBase.sol";
+
+// solhint-disable func-name-mixedcase
+
+abstract contract LidoGatewayManager is ScrollGatewayBase {
```

Comment:
Would it be possible to use OZ's [`AccessControlEnumerable`](https://docs.openzeppelin.com/contracts/4.x/api/access#AccessControlEnumerable) or at least `AccessControl`?

The idea is to have a full-fledged ACL system to allow:

- enabling deposits/withdrawals by the DAO
- pausing deposits/withdrawals by the DAO AND emergency break msigs

See also the proposed deployment params and ACL setup attached to this review :pray: 

Similar cases:
- Mantle: https://research.lido.fi/t/wsteth-deployment-on-mantle/5991/4
- zkSync Era: https://research.lido.fi/t/wsteth-deployment-on-zksync/5701/7

Reply:
The contract is used to upgrade an existing lido gateway. I'm afraid it is not possible.

Reply:
I see, then maybe allow multiple pause entities at least?

Reply:
@zimpha @TheDZhon What do you think about using [the Lido-modified unstructured version](https://github.com/lidofinance/lido-dao/blob/master/contracts/0.8.9/utils/access/AccessControlEnumerable.sol) here?

Reply:
Good catch!

Reply:
good catch